### PR TITLE
Upgrade lab environment

### PR DESCRIPTION
**Pull request by Peter9192**
This PR constitutes a major update of the environment.

- Inherit from newer jupyter/datascience-notebook
- Updated jupyter hub & jupyter lab versions
- Added explicit dependency of matplotlib
- Removed some packages as they are now part of base- or scipy-nb
- Removed some packages as they were installed as dependencies of others
- Moved all pip-installs to conda
- Add esmvalcore and esmvaltool-python
- Upgrade several packages to newer versions
- Disable not-working jupyter-thredds plugin

### CODE REVIEW

**Comment by JaroCamphuijsen** [comment]
@Peter Kalverla it seems that esmvaltool is installed and works in this docker container, however whenever I start ESMValTool in a terminal of the lab environment I get an error message:
`ERROR 1: PROJ: proj_create_from_database: Open of /opt/conda/share/proj failed`

Which is awkward since this folder exists and is not restricted to certain users. 

Probably a problem with the installation of proj. The error doesn't directly break esmvaltool, however I didn't try running a recipe with data yet. 

---------------------

*UPDATE*
To get rid of this error do:
`conda init bash`
open a new shell
`conda activate base`

I think we need to adapt the docker file still a bit to activate the environment in the right way or fix some environment paths and permissions.

**Comment by Peter9192** [comment]
> I think we need to adapt the docker file still a bit to activate the environment in the right way or fix some environment paths and permissions.

Yes, I also noticed this behaviour on the local build of the docker container (also the old version that I started with), but it is not necessary on the server. So I figured this is configured somewhere else, outside of the docker file. Let's look into that a bit more.

File: Dockerfile [hunk 1]
```diff
@@ -48,56 +48,43 @@ USER $NB_UID
 # - installing ncl under Conda messes up the notebook; for example, the R kernel disappears from the menu
 # - Downgrade Matplotlib to 2.2 for Iris
```
Comment (bouweandela) [inline Dockerfile]: This note is outdated

**Comment by Peter9192** [comment]
Setting `ENV PROJ_LIB=/opt/conda/share/proj` solves the problem, but note that this is only because on a local build, the conda base env isn't activated by default. On the current jupyterlab server, the conda base env *is* activated by default, and there the environment variable *has been* set. So it should probably not be necessary to set this explicitly outside the conda env.

Same goes for `ENV_UDUNITS2_XML_PATH`

**Review by JaroCamphuijsen — APPROVED** [review]

**Pull request merged**